configure.ac

```diff
@@ -118,6 +118,15 @@ DOSEMU_LDFLAGS="-Wl,-warn-common -pthread"
 
 AC_CHECK_LIB(m, pow)
 
+AC_CHECK_LIB(readline, main)
+if test "$ac_cv_lib_readline_main" = "yes"; then
```

Comment:
The problem with this code is that
AC_CHECK_LIB() automatically assigns
`-lreadline` to `LIBS`. But you don't use
it and add a `ac_cv_lib_readline_main`
hack and another variable.
There are a few ways around this.
First, you can use "action" arguments of
`AC_CHECK_LIB()`. In this case, IIRC,
it will not touch `LIBS` and you will not
touch `ac_cv_lib_readline_main`.
Another possibility is to move `dosdebug`
to plugins - in this case it will have its own
configure. If we want a modular build, I
believe this possibility is better, but I won't
insist on you doing that right now.

Reply:
Yes I noticed the LIBS had -lreadline appended. I did try adding a 'true' action to avoid that but then I found I lost the automatic definition of HAVE_LIBREADLINE and to define that I needed a template. This all seemed like a lot of hoops just to avoid the -lreadline in LIBS. I can try again, but as you can tell, I'm a novice with autoconf.

Reply:
But you really _must_ avoid `-lreadline` in
libs, because otherwise `dosemu.bin` will be
linked to libreadline. And debhelper will see
that and write a dependency. Of course it is
not all critical, but definitely a rust.
Perhaps the only clean solution is to move
it to plugins. To create a new plugin, you
need to get `configure.ac`, `Makefile.conf.in`
and `config/plugin_config.h` from `gpm`
plugin (it is a simplest plugin and I always
use it as a base) and do some manual
adjustments. Please take a look into these
3 files: 2 of them consists of just one line!
And `configure.ac` consists of 7 lines.
So its really all that simple, just take a look
before saying "I am not familiar with that". :)
If you want the plugin to be built by default,
you add it to `plugin_list`.
Or use `AH_TEMPLATE`.

Reply:
Okay I tried harder :smile: . The action method works fine now. I still have a problem with the debian package build though. I see this
~~~
/tmp/cc7w2qUe.o: In function `db_completion':
/clients/common/dosemu2.git/build/../src/tools/debugger/dosdebug.c:294: undefined reference to `rl_completion_matches'
/clients/common/dosemu2.git/build/../src/tools/debugger/dosdebug.c:299: undefined reference to `rl_attempted_completion_over'
/clients/common/dosemu2.git/build/../src/tools/debugger/dosdebug.c:299: undefined reference to `rl_attempted_completion_over'
/tmp/cc7w2qUe.o: In function `handle_dbg_input':
/clients/common/dosemu2.git/build/../src/tools/debugger/dosdebug.c:386: undefined reference to `rl_on_new_line'
/clients/common/dosemu2.git/build/../src/tools/debugger/dosdebug.c:387: undefined reference to `rl_redisplay'
/clients/common/dosemu2.git/build/../src/tools/debugger/dosdebug.c:386: undefined reference to `rl_on_new_line'
/clients/common/dosemu2.git/build/../src/tools/debugger/dosdebug.c:387: undefined reference to `rl_redisplay'
/tmp/cc7w2qUe.o: In function `handle_console_input':
/clients/common/dosemu2.git/build/../src/tools/debugger/dosdebug.c:331: undefined reference to `add_history'
/tmp/cc7w2qUe.o: In function `main':
/clients/common/dosemu2.git/build/../src/tools/debugger/dosdebug.c:477: undefined reference to `rl_readline_name'
/clients/common/dosemu2.git/build/../src/tools/debugger/dosdebug.c:480: undefined reference to `rl_attempted_completion_function'
/clients/common/dosemu2.git/build/../src/tools/debugger/dosdebug.c:483: undefined reference to `rl_callback_handler_install'
/clients/common/dosemu2.git/build/../src/tools/debugger/dosdebug.c:485: undefined reference to `rl_instream'
/clients/common/dosemu2.git/build/../src/tools/debugger/dosdebug.c:486: undefined reference to `rl_outstream'
/clients/common/dosemu2.git/build/../src/tools/debugger/dosdebug.c:553: undefined reference to `rl_crlf'
/clients/common/dosemu2.git/build/../src/tools/debugger/dosdebug.c:554: undefined reference to `rl_callback_handler_remove'
/clients/common/dosemu2.git/build/../src/tools/debugger/dosdebug.c:508: undefined reference to `rl_callback_read_char'
collect2: error: ld returned 1 exit status
Makefile:23: recipe for target 'dosdebug' failed
make[3]: *** [dosdebug] Error 1
make[3]: Leaving directory '/clients/common/dosemu2.git/build/src/tools/debugger'
/clients/common/dosemu2.git/build/../src/arch/linux/Makefile.main:118: recipe for target 'tools/debugger' failed
make[2]: *** [tools/debugger] Error 2
make[2]: Leaving directory '/clients/common/dosemu2.git/build/src'
Makefile:24: recipe for target 'default' failed
make[1]: *** [default] Error 2
make[1]: Leaving directory '/clients/common/dosemu2.git/build'
dh_auto_build: make -j1 returned exit code 2
debian/rules:6: recipe for target 'binary' failed
make: *** [binary] Error 2
dpkg-buildpackage: error: fakeroot debian/rules binary gave error exit status 2
debuild: fatal error at line 1376:
dpkg-buildpackage -rfakeroot -D -us -uc -i -b failed
Makefile:56: recipe for target 'deb' failed
make: *** [deb] Error 29
~~~

Why doesn't the `Depends: ${misc:Depends}, ${shlibs:Depends}` line include readline for me?